redis/src/cluster_async/mod.rs

```diff
@@ -718,6 +731,69 @@ where
         }
     }
 
+    async fn periodic_topology_check(inner: Arc<InnerCore<C>>, interval_duration: Duration) {
+        loop {
+            #[cfg(feature = "tokio-comp")]
+            tokio::time::sleep(interval_duration).await;
+            #[cfg(all(not(feature = "tokio-comp"), feature = "async-std-comp"))]
+            async_std::task::sleep(interval_duration).await;
+            let read_guard = inner.conn_lock.read().await;
+            if read_guard.get_current_topology_hash().is_none() {
+                drop(read_guard);
+                continue;
+            }
+            drop(read_guard);
+
+            let retry_strategy = ExponentialBackoff {
+                initial_interval: DEFAULT_REFRESH_SLOTS_RETRY_INITIAL_INTERVAL,
+                max_interval: DEFAULT_REFRESH_SLOTS_RETRY_TIMEOUT,
+                ..Default::default()
+            };
+            let topology_check_res = retry(retry_strategy, || {
+                Self::check_for_topology_diff(inner.clone()).map_err(Error::from)
+            })
+            .await;
+            if topology_check_res.is_ok() && topology_check_res.unwrap() {
+                let _ = Self::refresh_slots(inner.clone(), DEFAULT_NUMBER_OF_REFRESH_SLOTS_RETRIES)
```

Comment:
TODO: 
1. Add a flag that indicates if refresh slots is already in progress and we can return with no-op.
2. Call refresh_slots_with_retries instead